WordPress/src/main/java/org/wordpress/android/ui/media/MediaGridFragment.java

```diff
@@ -437,20 +476,53 @@ public void setFilterVisibility(int visibility) {
         }
     }
 
-    private void setEmptyViewVisible(boolean visible) {
-        setEmptyViewVisible(visible, -1);
+    private void updateEmptyView(MessageType messageType) {
+        if (mGridAdapter.getDataCount() == 0) {
+            showEmptyViewWithMessage(messageType);
+        } else {
+            hideEmptyView();
+        }
     }
 
-    private void setEmptyViewVisible(boolean visible, int messageId) {
-        if (visible) {
-            mGridView.setVisibility(View.GONE);
-            mEmptyView.setVisibility(View.VISIBLE);
-            if (messageId != -1) {
-                mEmptyViewTitle.setText(getResources().getString(messageId));
+    private void showEmptyViewWithMessage(MessageType messageType) {
```

Comment:
The other implementations of updateEmptyView handle everything contained in showEmptyViewWithMessage. For consistency and simplicity I'd recommend doing the same here.